HW06.Rmd

```diff
@@ -1,92 +1,172 @@
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+# I added STK38, DIS3L2, and PIEZO1
+
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2", "STK38", "DIS3L2", "PIEZO1")
+
+
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+
+# generate new dataframe with only the H3K27M data and tidy
+H3K27M <- GBM.genes.of.interest %>%
+  dplyr::select(gene_symbol, TPM...2:TPM...24) %>% # select the columns that correspond 
+  gather(key = "TPM_number", value = "value", 2:24, -gene_symbol) %>% # gather TPM values into a column
+  spread(key = gene_symbol, value = value) # spread TPM values by gene_symbol
+
+
+# do the same for WT data
+WT <- GBM.genes.of.interest %>%
+  dplyr::select(gene_symbol, TPM...25:TPM...46) %>%
+  gather(key = "TPM_number", value = "value", 2:23, -gene_symbol) %>%
+  spread(key = gene_symbol, value = value)
+
+# generate an empty vector for storing the p values from the ttest and a vector with the names of the genes
+p_value <- c()
+gene_name <- colnames(H3K27M)
+
+# use a for loop to perform a ttest for each gene and store the p value in the empty vector
+for (i in 2:ncol(H3K27M)) {
+  p_value[i] <- t.test(H3K27M[,i], WT[,i])$p.value
+}
+
+# generate a new dataframe that combines gene name and p value
+ttest_results <- data.frame(gene_name, p_value)
+
+# remove the first row of this data frame
+ttest_results_final <- ttest_results[-1,]
+
+
+
+#print out the t-test results
+knitr::kable(ttest_results_final, format = "markdown", col.names = c("Gene", "p-value"), align = "c")
```

Comment:
I got inspration from your homework in how to generate the table using a `p_value` vector. Thanks!